### PR TITLE
CMake improvements

### PR DESCRIPTION
**Pull request by cor3ntin**

### CODE REVIEW

File: viewer/CMakeLists.txt [hunk 1]
```diff
@@ -1,38 +1,39 @@
...
-		Viewer.h
-	)
-	install(FILES ${ENKI_VIEWER_HDR}
-		DESTINATION include/viewer/
-	)
-	install(TARGETS enkiviewer ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
-                                   LIBRARY DESTINATION ${LIB_INSTALL_DIR})
-endif ()
+add_library(enkiviewer
+    Viewer.cpp
```
Comment (stephanemagnenat) [inline viewer/CMakeLists.txt]: Inconsistent use of tab and space

**Comment by stephanemagnenat** [comment]
To which version of CMake does this PR makes Enki depend?

**Comment by cor3ntin** [comment]
@stephanemagnenat cmake version has not changed ( 2.8.11 )

**Comment by stephanemagnenat** [comment]
This PR makes Qt5 mandatory, right?

**Comment by cor3ntin** [comment]
@stephanemagnenat `enki` does not require Qt5, `enkiviewer` does. It's consistent with the previous behavior.

**Comment by stephanemagnenat** [comment]
Yes of course, but I do not understand how the patch keeps this behaviour, as there is no test in the main `CMakeLists.txt` to selectively include `viewer/`, and the viewer unconditionally depends on Qt5. The condition exists when using the config file, but as far as I understand it has been removed from the main file. But maybe I missed something?

**Comment by cor3ntin** [comment]
Yeah, you are right, I will fix that.

File: viewer/CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,12 @@
+find_package(Qt5 5  COMPONENTS Core Gui Widgets OpenGL)
```
Comment (stephanemagnenat) [inline viewer/CMakeLists.txt]: What is the aim of the 5 after Qt5? Is it a typo?

File: CMakeLists.txt [hunk 3]
```diff
@@ -7,12 +7,6 @@ set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/CMakeModules)
...
 enable_testing()
 
-# check for Qt
-find_package(Qt5Core)
-find_package(Qt5Gui)
-find_package(Qt5Widgets)
-find_package(Qt5OpenGL)
-
 # check for SDL2
 find_package(SDL2)
```
Comment (stephanemagnenat) [inline CMakeLists.txt]: For the sake of consistency, if the checks for Qt are removed from the top-level, this check for SDL2 should as well, and be put in `examples/playground`.

**Comment by stephanemagnenat** [comment]
I added two comments, once they are fixed I can merge.

File: CMakeLists.txt (hunk 3 above)
Comment (cor3ntin) [reply CMakeLists.txt]: I will move back the Qt5 check to the top-level file, it makes more sense

**Comment by cor3ntin** [comment]
@stephanemagnenat In case you didn't get notified. Thanks !

**Pull request merged**